### ISSUE TITLE
Consider methods for increasing update frequency of ontology updates (in Minerva)

### ISSUE THREAD

**Issue by kltm** [geneontology/noctua#737]
Currently, the SOP for bringing production minerva up to date is to bring it down at one of our twice-a-month outages, destroy the ontology journal and restart minerva (which rebuilds the ontology journal as it is no longer there). At a request from @vanaukenk we looked into possible ways of speeding things up or doing them more frequently.

(Currently leaving out of joining a project as still under discussion.)

There are three general routes so far proposed.

1) Do what we currently do, but more often. This would be irritating and fiddly, but straightforward and doable. It would require more outages and more manual time.

2) Get minerva to update ontologies on its own (internally). While this is not currently available, @balhoff believes that we're currently in a better position to move in this direction than we were in the past due to more strict internal separations. It may be worthwhile to flesh out a bit more what that might mean and how hard it would be to do. This would likely be a mini-project on its own, but would cut down on overall maintenance and fiddliness.

3) Shorten the minerva reboot cycle so that it can be done more easily and frequently. We talked a fair bit about what would happen if we just built a new ontology journal, turned off minerva, switch the new ontology journal in, and turned it on again. @balhoff believes that this should be fine, but we could move on and test that on noctua-dev. That said, on further consideration, I'm not sure this save much time as I believe that the intial minerva "scan" of the files is not currently avoidable on restart. I guess I would ask @balhoff , similar to the above, what purpose the scan is currently serving and if there is a reasonable way to truncate it?

**Comment by balhoff** [comment]
I thought we were actually talking about model load yesterday in addition to the ontology journal. I don't think there is any reason to reload all the models from files between restarts, unless an offline bulk modification on the files is performed. Otherwise just use the same model journal. No need to delete it and re-import models.

**Comment by kltm** [comment]
@balhoff Naturally, but I think this is just the way that minerva works right now. If I kill minerva and restart it, it goes through a scanning process of all of the models independent of rebuilding the ontology journal. This scan portion is what is taking most all of the time, not the ontology part--if we just juggle journals, there would be little time savings (although good to know that we can and might be otherwise useful).

**Comment by kltm** [comment]
@balhoff @cmungall Okay, with slightly red cheeks, there is indeed a bundle that run things and I have re-familiarized myself with how this actually works. Decomposing that, we have:

* "destruction of ontology journal"
* "destruction of journal"
* "re-importation" `java -Xmx128G -jar ../minerva/minerva-cli/bin/minerva-cli.jar --import-owl-models -j /srv/noctua/blazegraph.jnl  -f ~/local/src/git/noctua-models/models/` <- this is re-creating the journal and the "scan"; in the past we ran SPARQL changes after this step
* "startup" `java -Xmx128G -cp ../minerva/minerva-cli/bin/minerva-cli.jar org.geneontology.minerva.server.StartUpTool --use-golr-url-logging --use-request-logging --arachne -g http://skyhook.berkeleybop.org/go-lego-reacto.owl --set-important-relation-parent http://purl.obolibrary.org/obo/LEGOREL_0000000 --port 6800 --ontojournal /srv/noctua/blazegraph-go-lego-reacto-neo.jnl -f /srv/noctua/blazegraph.jnl --export-folder /home/swdev/local/src/git/noctua-models/models/ --golr-labels http://noctua-golr.berkeleybop.org --golr-seed http://golr-aux.geneontology.io/solr --skip-class-id-validation` <- this step would download/recreate the ontology journal and properly start up

The scanning occurs in "re-importation" there.

Okay, I just tried a minerva recompile and restart (just "start") on noctua-dev and it seems to have been perfectly happy. While we'd still need to have large windows available for imports and updates, the week-to-week could just be "destruction of ontology journal" + "startup". @vanaukenk We can give this a try at the next periodic outage.
